src/main/java/com/uber/cadence/internal/sync/WorkflowThreadImpl.java

```diff
@@ -92,13 +99,26 @@ public void run() {
       ContextThreadLocal.setContextPropagators(this.contextPropagators);
       ContextThreadLocal.propagateContextToCurrentThread(this.propagatedContexts);
 
-      try {
+      // Set up an opentracing span
```

Comment:
Changing the code here for each new propagator is not going to scale. Can we do it in the "propagateContexttoTheCurrentThread"? Otherwise we need to extend the propagator API to avoid such "workarounds".

Reply:
Yeah, I wasn't super happy about having to do this.  Unlike Go, there's nowhere to drop the span information outside the context of an active span.  

One option I suppose would be to implement some sort of `ContextPropagator.cleanup()` method that got invoked at the end of a Workflow/Activity - in that case we could start/end the span solely inside the propagator. 

Thoughts?

Reply:
@mfateev OK - I went with option 1, and now there's no OpenTracing-specific code in the core library except where we automatically add an OpenTracingContextPropagator if one doesn't exist.  Would be curious to get your take on that decision